lib/Conversion/PassDetail.h

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_LIB_CONVERSION_PASSDETAIL_H
```

Comment:
It does not seem like we should have .h files under lib..should this go to include/conversion/passes.h?

Reply:
Also lib/Conversion/TTNNToEmitC/PopulatePatterns.h

Reply:
Those were my thoughts too, but since we agreed to model this approach the same way torch-mlir does, I copied their structure. Example: https://github.com/llvm/torch-mlir/tree/main/lib/Conversion/TorchToLinalg

Reply:
Agreed where? I went through torch-mlir repo a few times, and I noticed they didn't align with the canon MLIR recommendations on a few occasions so far. Not saying they are doing it wrong, but it does not mean we have to copy. Take a look at LLVM repo https://github.com/llvm/llvm-project/tree/main/mlir/lib/Conversion.

Reply:
I remember someone mentioning in one of the uhuru meetings that some of the original MLIR folks were now working on torch-mlir, and that we should follow their steps with regards to code organization.

So I took @nsmithtt's suggestion ([here](https://github.com/tenstorrent/tt-mlir/issues/94)) at face value.

I just took a short glance at LLVM repo, it looks like all of their `.h` files in `include` dir are very simple and only serve the purpose of registering stuff, same as here. In `lib` folder though, they seem to pack the equivalent of `PopulatePatterns.h` in the original `.cpp`. So it is different to torch-mlir, but torch-mlir code looks much more readable to me.

To me personally, it seems a bit strange to separate `.h` and `.cpp` files between these two folders IFF they're not a standardized MLIR mechanism.

All that said, it's up to us to decide what we want since there's no wrong/right. Thoughts?

Reply:
Yeah maybe torch-mlir doesn't follow things as closely as expected, the project is relatively old by MLIR standards, maybe there were decisions they made before there was an official MLIR way to do it.

I agree with @nobradovictt, let's look to the llvm-project first and torch-mlir second.

All that said, there a 66 headers under `llvm-project/mlir/lib` so I think this is an acceptable way to define internal interfaces.

Reply:
Sounds good, I'll leave it as is.